safe_transaction_service/history/indexers/safe_events_indexer.py

```diff
@@ -485,3 +478,137 @@ def _process_decoded_element(
+                        elif (
+                            event_position == 1
+                            and events[0]["event"] == "ProxyCreation"
+                        ):
+                            # ProxyCreation first and SafeSetup later
+                            proxy_creation_event = events[1]
+                    else:
+                        # Proxy was created in previous blocks.
+                        proxy_creation_event = None
+                        # TODO store decoded ProxyCreation event to get the singleton address
+
+                    # Generate InternalTx and internalDecodedTx for SafeSetup event
+                    setup_trace_address = (
+                        f"{str(proxy_creation_event['logIndex'])},0"
+                        if proxy_creation_event
+                        else str(setup_event["logIndex"])
+                    )
+                    singleton = (
+                        proxy_creation_event["args"].get("singleton")
+                        if proxy_creation_event
+                        else NULL_ADDRESS
+                    )
+                    internal_tx = self._get_internal_tx_from_decoded_event(
+                        setup_event,
+                        to=singleton,
+                        trace_address=setup_trace_address,
+                        call_type=EthereumTxCallType.DELEGATE_CALL.value,
+                    )
+                    # Generate InternalDecodedTx for SafeSetup event
+                    setup_args = dict(event["args"])
+                    setup_args["payment"] = 0
+                    setup_args["paymentReceiver"] = NULL_ADDRESS
+                    setup_args["_threshold"] = setup_args.pop("threshold")
+                    setup_args["_owners"] = setup_args.pop("owners")
+                    internal_tx_decoded = InternalTxDecoded(
+                        internal_tx=internal_tx,
+                        function_name="setup",
+                        arguments=setup_args,
+                    )
+                    internal_txs.append(internal_tx)
+                    internal_decoded_txs.append(internal_tx_decoded)
+                elif event["event"] == "ProxyCreation":
+                    proxy_creation_event = event
+                    # Generate InternalTx for ProxyCreation
+                    internal_tx = self._get_internal_tx_from_decoded_event(
+                        proxy_creation_event,
+                        contract_address=proxy_creation_event["args"].get("proxy"),
+                        tx_type=InternalTxType.CREATE.value,
+                        call_type=None,
+                    )
+                    internal_txs.append(internal_tx)
+                else:
+                    logger.error(f"Event is not a Safe creation event {event['event']}")
+
+        with transaction.atomic():
+            InternalTx.objects.bulk_create(internal_txs)
+            InternalTxDecoded.objects.bulk_create(internal_decoded_txs)
```

Comment:
What happens if they are already inserted?

Reply:
Raise an exception.
Ignoring conflicts in such case would fix the issue, https://github.com/safe-global/safe-transaction-service/pull/2194/commits/5eeabd4683d121afcf6d0247dc71152282efa1a6

Reply:
@Uxio0 `ignore_conflicts` fail because `InternalDecodedTx` have relation with `internal_tx`, basically the ORM try to prevent lose data.
We can use instead of that update_conflicts, but I'm concerned because I just discovered that bulk_create does not call save, and in that case post_save will not be fired. 
Will this affect our signals handle?

Reply:
`bulk_create` does in case we inherit our BulkCreateMixin

Reply:
I did a workaround in case that would be a collision inserting the same instance, shouldn't be any common because the bulk create is in atomic section and we are checking if the contract were indexed, basically if the `setup` and `proxyevent` were correctly inserted, shouldn't reach the insert section if would be indexed again.